### PR TITLE
build abi3 wheels

### PR DESCRIPTION
**Pull request by kdashg**
- change c extensions to use the limited ABI
 - update `setup.py` to build abi3 wheels
 - add Python 3.11 to build matrix
 - rebased 2023may08 from #342

### CODE REVIEW

**Comment by kdashg** [comment]
#342 appears to have stalled, so here's a rebase!

**Comment by kdashg** [comment]
@jlaine PTAL!

**Comment by jlaine** [comment]
I'd love to get this merged, but CI seems to be in a weird place right now..

File: src/aioquic/_buffer.c [hunk 1]
```diff
@@ -5,6 +5,12 @@
 
 #define MODULE_NAME "aioquic._buffer"
 
+// https://foss.heptapod.net/pypy/pypy/-/issues/3770
```
Comment (jlaine) [inline src/aioquic/_buffer.c:8]: It looks like this issue was fixed, can we drop this?

**Comment by jlaine** [comment]
OK I've made a couple of changes:

- `wheel` needed to be added to build requirements in `pyproject.toml` since we import it from our `setup.py`
- the workaround for pypy was not needed anymore
- the imports in `setup.py` were not sorted correctly

**Pull request merged**